File: cpp/sfo_15_number_of_1_bits_s2/sfo_15_number_of_1_bits_s2.cpp
/*
* File: sfo_15_number_of_1_bits_s2.cpp
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

#include "../include/include.hpp"

// ===== Solution Code =====
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int res = 0;
        while(n != 0) {
            res++;
            n &= n - 1;
        }
        return res;
    }
};

int main() {
    // ======= Test Case =======
    uint32_t n = 11;
    // ====== Driver Code ======
    Solution* slt = new Solution();
    int res = slt->hammingWeight(n);
    cout << res << endl;
    
    return 0;
}

File: sword_for_offer/codes/python/sfo_15_number_of_1_bits_s2.py
'''
File: sfo_15_number_of_1_bits_s2.py
Created Time: 2021-12-09
Author: Krahets (krahets@163.com)
'''

from include import *

# ===== Solution Code =====
class Solution:
    def hammingWeight(self, n: int) -> int:
        res = 0
        while n:
            res += 1
            n &= n - 1
        return res

# ======= Test Case =======
n = 11
# ====== Driver Code ======
slt = Solution()
res = slt.hammingWeight(n)
print(res)
